WcaOnRails/app/models/committee_position.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+class CommitteePosition < ActiveRecord::Base
+  belongs_to :committee
+  has_many :team_members
+  has_many :current_members, -> { current }, class_name: "TeamMember"
+
+  SENIOR_DELEGATE = "senior-delegate"
+  DELEGATE = "delegate"
+  CANDIDATE_DELEGATE = "candidate-delegate"
+
+  MAX_NAME_LENGTH = 50
+  VALID_NAME_RE = /\A[[:alnum:] -]+\z/
+  MAX_SLUG_LENGTH = 50
+  VALID_SLUG_RE = /\A[[:alnum:]-]+\z/
+  MAX_DESCRIPTION_LENGTH = 255
+
+  validates :name, presence: true, uniqueness: {scope: :committee_id}, length: { maximum: MAX_NAME_LENGTH }, format: { with: VALID_NAME_RE, message: "must contain alpanumric characters and spaces only" }
+  validates :slug, presence: true, uniqueness: {scope: :committee_id}, length: { maximum: MAX_SLUG_LENGTH }, format: { with: VALID_SLUG_RE, message: "must contain alpanumric characters and dashes only" }
+  validates :description, presence: true, length: { maximum: MAX_DESCRIPTION_LENGTH }
+  before_validation :compute_slug
+  private def compute_slug
+    self.slug ||= name.parameterize
+  end
```

Comment:
Are you sure? What about [this](https://github.com/pingskills/worldcubeassociation.org/blob/f9f7f423ef6cdfe08a8d4e94958369b6c57f452e/WcaOnRails/app/controllers/committee_positions_controller.rb#L62-L64).